### PR TITLE
Feature/id 353/option to show and hide incidents

### PR DESCRIPTION
**Pull request by saletG08**
### Authors:
Juan Carlos Hidalgo @JCarlosHidalgoSosa 
Gaston Gutierrez @GastonGutierrezC 
Victor Villca @victor0villca 
Luiggy Mamani @luiggymamani 
Salet Gutierrez @saletG08 

### What?
A screen was created where you have the options if you want to show the incidents or generate incidents, we implemented the show incidents where it will take us to another screen and we can select which incidents we want to show.

### How?
Implementation of the view
Implementation of images and xml files.
Style implentation.
Implementation of strings.

### Why?
The task was done so that when using the application the person can see the incidents that exist on the map, either Street Incidents, Natural Disasters or Traffic, according to what is selected will be displayed and will be hidden if you stop selecting.

### Code Reviewers/Testers

Team A: @Karvlox 
Team B: @JeferssonCL 
Team C: @CatCoder36 

### Trainer (General Reviewer):
@mmoralesv

### CODE REVIEW

File: app/src/main/java/com/isc/hermes/MainActivity.java [hunk 1]
```diff
@@ -469,4 +497,9 @@ public boolean onNavigationItemSelected(@NonNull MenuItem item) {
         }
         return true;
     }
+
+    public void acceptNaturalDisasters(){
```
Comment (Karvlox) [inline app/src/main/java/com/isc/hermes/MainActivity.java:502]: This method does not have Javadoc

File: app/src/main/java/com/isc/hermes/controller/IncidentsGetterController.java [hunk 2]
```diff
@@ -39,27 +43,33 @@ public IncidentsGetterController() {
...
      *
-     * @param mapboxMap The Mapbox map instance.
      */
-    public void getNearIncidentsWithinRadius(MapboxMap mapboxMap, Context context) {
+    public void getNearIncidentsWithinRadius(Context context) {
         incidentsPointView = IncidentPointVisualizationController.getInstance(mapboxMap, context);
+        loadNearIncidents(context);
+    }
+
+    private void loadNearIncidents(Context context){
```
Comment (Karvlox) [inline app/src/main/java/com/isc/hermes/controller/IncidentsGetterController.java:63]: This method does not have Javadoc

File: app/src/main/java/com/isc/hermes/controller/MapPolygonController.java [hunk 3]
```diff
@@ -92,4 +94,36 @@ public void deleteMarks() {
             mapboxMap.removeMarker(marker);
         }
     }
+
+    public MapboxMap getMapboxMap() {
```
Comment (Karvlox) [inline app/src/main/java/com/isc/hermes/controller/MapPolygonController.java:103]: This method does not have Javadoc

File: app/src/main/java/com/isc/hermes/controller/MapPolygonController.java [hunk 4]
```diff
@@ -92,4 +94,36 @@ public void deleteMarks() {
...
+        return mapboxMap;
+    }
+
+    public List<Point> getPolygonPoints() {
```
Comment (Karvlox) [inline app/src/main/java/com/isc/hermes/controller/MapPolygonController.java:112]: This method does not have Javadoc

File: app/src/main/java/com/isc/hermes/controller/PolygonOptionsController.java [hunk 5]
```diff
@@ -62,6 +77,40 @@ private void handleCancelButtonClick() {
...
+
+            @SuppressLint("StaticFieldLeak")
+            @Override
+            protected void onPostExecute(Integer responseCode) {
+                handleUploadResponse(responseCode);
+            }
+        };
+        task.execute();
+    }
+    private void handleUploadResponse(Integer responseCode) {
```
Comment (Karvlox) [inline app/src/main/java/com/isc/hermes/controller/PolygonOptionsController.java:105]: This method does not have Javadoc

File: app/src/main/java/com/isc/hermes/controller/PolygonOptionsController.java [hunk 6]
```diff
@@ -62,6 +77,40 @@ private void handleCancelButtonClick() {
...
+            }
+
+            @SuppressLint("StaticFieldLeak")
+            @Override
+            protected void onPostExecute(Integer responseCode) {
+                handleUploadResponse(responseCode);
+            }
+        };
+        task.execute();
+    }
```
Comment (Karvlox) [inline app/src/main/java/com/isc/hermes/controller/PolygonOptionsController.java:98]: Missing line break

File: app/src/main/java/com/isc/hermes/controller/PolygonOptionsController.java [hunk 7]
```diff
@@ -62,6 +77,40 @@ private void handleCancelButtonClick() {
...
+        System.out.println("EL PEPEPE");
+        if (responseCode == HttpURLConnection.HTTP_OK) {
+            Toast.makeText(context, R.string.natural_disaster_uploaded, Toast.LENGTH_SHORT).show();
+
+        } else {
+            Toast.makeText(context, R.string.natural_disaster_not_uploaded, Toast.LENGTH_SHORT).show();
+        }
+    }
+
+    private Integer uploadNaturalDisasterDataBase(){
```
Comment (Karvlox) [inline app/src/main/java/com/isc/hermes/controller/PolygonOptionsController.java:121]: This method does not have Javadoc

File: app/src/main/java/com/isc/hermes/view/GenerateRandomIncidentView.java [hunk 8]
```diff
@@ -35,26 +39,27 @@ public GenerateRandomIncidentView(AppCompatActivity activity){
...
+                = ArrayAdapter
+                .createFromResource(
+                        activity, R.array.spinner_items_incidents_generate
+                        , android.R.layout.simple_spinner_item
+                );
         adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
         spinner.setAdapter(adapter);
     }
 
+    private void initGenerateIncidentButtonFunctionality() {
```
Comment (Karvlox) [inline app/src/main/java/com/isc/hermes/view/GenerateRandomIncidentView.java]: This method does not have Javadoc

File: app/src/main/java/com/isc/hermes/controller/TrafficAutomaticFormController.java [hunk 9]
```diff
@@ -104,7 +104,7 @@ public int uploadTrafficDataBase(){
         String dateCreated = IncidentsUtils.getInstance().generateCurrentDateCreated();
         String deathDate = IncidentsUtils.getInstance().addTimeToCurrentDate(getTrafficTime());
         String coordinates = TrafficUploader.getInstance().getCoordinates();
-        String JsonString = TrafficUploader.getInstance().generateJsonIncident(id,getTrafficType(20,50),"Traffic Test",dateCreated, deathDate , GeometryType.LINE_STRING.getName(),coordinates);
+        String JsonString = TrafficUploader.getInstance().generateJsonIncident(getTrafficType(20,50),"Traffic Test",dateCreated, deathDate , GeometryType.LINE_STRING.getName(),coordinates);
```
Comment (CatCoder36) [inline app/src/main/java/com/isc/hermes/controller/TrafficAutomaticFormController.java]: The "reason" parameter received by this method should be set only as traffic and not traffic test.

**Review by JeferssonCL — APPROVED** [review]
No comments

**Review by Karvlox — APPROVED** [review]

**Review by JeferssonCL — APPROVED** [review]

**Review by CatCoder36 — APPROVED** [review]
Good job

File: app/src/main/java/com/isc/hermes/MainActivity.java (hunk 1 above)
Comment (saletG08) [reply app/src/main/java/com/isc/hermes/MainActivity.java:502]: done

File: app/src/main/java/com/isc/hermes/controller/IncidentsGetterController.java (hunk 2 above)
Comment (saletG08) [reply app/src/main/java/com/isc/hermes/controller/IncidentsGetterController.java:63]: done

File: app/src/main/java/com/isc/hermes/controller/MapPolygonController.java (hunk 3 above)
Comment (saletG08) [reply app/src/main/java/com/isc/hermes/controller/MapPolygonController.java:103]: done

File: app/src/main/java/com/isc/hermes/controller/MapPolygonController.java (hunk 4 above)
Comment (saletG08) [reply app/src/main/java/com/isc/hermes/controller/MapPolygonController.java:112]: done

File: app/src/main/java/com/isc/hermes/controller/PolygonOptionsController.java (hunk 5 above)
Comment (saletG08) [reply app/src/main/java/com/isc/hermes/controller/PolygonOptionsController.java:105]: done

File: app/src/main/java/com/isc/hermes/controller/PolygonOptionsController.java (hunk 6 above)
Comment (saletG08) [reply app/src/main/java/com/isc/hermes/controller/PolygonOptionsController.java:98]: It is necessary for a better view.

File: app/src/main/java/com/isc/hermes/controller/PolygonOptionsController.java (hunk 7 above)
Comment (saletG08) [reply app/src/main/java/com/isc/hermes/controller/PolygonOptionsController.java:121]: done

File: app/src/main/java/com/isc/hermes/view/GenerateRandomIncidentView.java (hunk 8 above)
Comment (saletG08) [reply app/src/main/java/com/isc/hermes/view/GenerateRandomIncidentView.java]: done

File: app/src/main/java/com/isc/hermes/controller/TrafficAutomaticFormController.java (hunk 9 above)
Comment (saletG08) [reply app/src/main/java/com/isc/hermes/controller/TrafficAutomaticFormController.java]: done

**Pull request merged**